### PR TITLE
Implement the `indirectly_comparable` concept

### LINKED ISSUE #444: Implement `indirectly_comparable` concept

### PR DESCRIPTION
**Pull request by miscco**
Fixes #444

### CODE REVIEW

File: libcudacxx/include/cuda/std/detail/libcxx/include/__iterator/indirectly_comparable.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  __indirectly_comparable_,
+  requires()(
+    requires(indirect_binary_predicate<_Rp, projected<_I1, _P1>, projected<_I2, _P2>>)
+  ));
+
+template <class _I1, class _I2, class _Rp, class _P1 = identity, class _P2 = identity>
+_LIBCUDACXX_CONCEPT indirectly_comparable =
+  _LIBCUDACXX_FRAGMENT(__indirectly_comparable_, _I1, _I2, _Rp, _P1, _P2);
+
+#endif // _LIBCUDACXX_STD_VER > 14
```
Comment (miscco) [inline libcudacxx/include/cuda/std/detail/libcxx/include/__iterator/indirectly_comparable.h:47]: In hindsight we might just simply go with
```suggestion
#if _LIBCUDACXX_STD_VER > 14

template <class _I1, class _I2, class _Rp, class _P1 = identity, class _P2 = identity>
_LIBCUDACXX_CONCEPT indirectly_comparable =
  indirect_binary_predicate<_Rp, projected<_I1, _P1>, projected<_I2, _P2>>;

#endif // _LIBCUDACXX_STD_VER > 14
```

It is only a single concept without any further requirements

**Comment by miscco** [comment]
ping

File: libcudacxx/include/cuda/std/detail/libcxx/include/__iterator/indirectly_comparable.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+#if defined(_LIBCUDACXX_USE_PRAGMA_GCC_SYSTEM_HEADER)
+#pragma GCC system_header
+#endif
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+#if _LIBCUDACXX_STD_VER > 17
+
+template <class _I1, class _I2, class _Rp, class _P1 = identity, class _P2 = identity>
```
Comment (senior-zero) [inline libcudacxx/include/cuda/std/detail/libcxx/include/__iterator/indirectly_comparable.h:30]: suggestion: I guess `_Rp` stands for Relational Predicate, which is a pedicate taking two or more arguments. In this case, the predicate can only take two arguments so we might consider `BinaryPredicate` or `Comp` to make the concept a bit more readable.

**Review by senior-zero — APPROVED** [review]

File: libcudacxx/include/cuda/std/detail/libcxx/include/__iterator/indirectly_comparable.h (hunk 2 above)
Comment (miscco) [reply libcudacxx/include/cuda/std/detail/libcxx/include/__iterator/indirectly_comparable.h:30]: Thanks a lot I am loving it

**Pull request merged**